TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMode.java

```diff
@@ -0,0 +1,36 @@
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+
+import org.firstinspires.ftc.teamcode.interfaces.Subsystem;
+import org.firstinspires.ftc.teamcode.subsystems.Drivetrain;
+import org.firstinspires.ftc.teamcode.util.SmartController;
+
+import java.util.ArrayList;
+
+@com.qualcomm.robotcore.eventloop.opmode.TeleOp (name = "Teste")
+public class TeleOpMode extends OpMode {
+
+    private ArrayList<Subsystem> Subsystems;
+    private SmartController driver;
+    private SmartController operator;
+    @Override
+    public void init() {
+        this.driver = new SmartController(gamepad1);
+        this.operator = new SmartController(gamepad2);
+        this.Subsystems = new ArrayList<Subsystem>();
+        this.Subsystems.add(Drivetrain.getInstance());
+
+        Subsystems.forEach(subsystem -> subsystem.initialize(hardwareMap, telemetry));
+    }
+
+    @Override
+    public void loop() {
+        Drivetrain.getInstance().arcadeDrive(-driver.getLeftStickY(), driver.getRightStickX());
```

Comment:
suggestion: every subsystem will need to execute a command. why not to implement on the interface?